pkg/apis/secondaryscheduler/v1/types.go

```diff
@@ -0,0 +1,66 @@
+package v1
+
+import (
+	operatorv1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// SecondaryScheduler is the Schema for the deschedulers API
+// +k8s:openapi-gen=true
+// +genclient
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+type SecondaryScheduler struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +required
+	Spec SecondarySchedulerSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status SecondarySchedulerStatus `json:"status"`
+}
+
+// SecondarySchedulerSpec defines the desired state of SecondaryScheduler
+type SecondarySchedulerSpec struct {
+	operatorv1.OperatorSpec `json:",inline"`
+
+	// Profiles sets which secondary scheduler strategy profiles are enabled
+	SchedulerConfig SecondarySchedulerConfig `json:"schedulerConfig"`
```

Comment:
We should discuss design for this. On one hand, I see providing `TargetLoadPacking` and `LoadVariationRiskBalancing` as easy ways to deploy those plugins

However, doing that will imply our support of those plugins which we don't want to do (at least not yet). I think from a design perspective this field should just be a configmap reference pointing to whatever custom config the user has defined (mentioned below that [the old Policy approach](https://docs.openshift.com/container-platform/4.1/nodes/scheduling/nodes-scheduler-default.html) would be good for this)

Reply:
These configmaps require updating the Prometheus tokens, etc., which are only required to be configured in openshift. In Kubernetes env, these plugins can be configured with Prometheus host without tokens.

I agree it should support any customized configmap.  I will change accordingly.

And could we still keep the TargetLoadPacking and LoadVariationRiskBalancing configmaps just in case users specify these in the secondary-scheduler cr but did not create these configmaps?

Reply:
> could we still keep the TargetLoadPacking and LoadVariationRiskBalancing as available options just in case users specify these in the secondary-scheduler cr but did not create these configmaps?

I would delegate the final call to PM, but imo adding these implies support for the actual plugins, which we do not want to officially provide yet. We can always add them later if we do end up officially supporting them

Reply:
@damemi , I changed the schedulerConfig field to any string that can take any configmap names defined by users and removed the TargetLoadPacking and LoadVariationRiskBalancing profile variables. I also add two configmap examples for these plugins under deploy folder.